tools/storage/tests/reap_worktrees_test.sh

```
@@ -0,0 +1,177 @@
+# REAP_SCRIPT lets the gate be pointed at an older copy of the script, so that "this suite is
+# RED before the fix and GREEN after" is something anyone can re-run rather than take on trust.
+SCRIPT="${REAP_SCRIPT:-$REPO/tools/storage/reap_worktrees.sh}"
+[ -f "$SCRIPT" ] || { echo "missing $SCRIPT"; exit 1; }
+
+# Resolve the sandbox to its PHYSICAL path. On macOS $TMPDIR both ends in a slash and is a
+# symlink into /private/var, and the script under test reports `pwd -P` paths — so an
+# unresolved expectation here would fail every case for a reason that has nothing to do with
+# the script's behaviour.
+SANDBOX="$(mktemp -d "${TMPDIR:-/tmp}/reapgate.XXXXXX")"
+SANDBOX="$(cd "$SANDBOX" && pwd -P)"
+cleanup() { [ -n "${SANDBOX:-}" ] && [ -d "$SANDBOX" ] && rm -rf "$SANDBOX"; }
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)